src/test/java/seedu/address/model/addresses/PublicAddressesCompositionTest.java

```diff
@@ -0,0 +1,391 @@
+package seedu.address.model.addresses;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.TypicalPublicAddresses.VALID_PUBLIC_ADDRESS_BTC_MAIN;
+import static seedu.address.testutil.TypicalPublicAddresses.VALID_PUBLIC_ADDRESS_BTC_SUB;
+import static seedu.address.testutil.TypicalPublicAddresses.VALID_PUBLIC_ADDRESS_ETH_MAIN;
+import static seedu.address.testutil.TypicalPublicAddresses.VALID_PUBLIC_ADDRESS_ETH_SUB;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import org.junit.jupiter.api.Test;
+
+public class PublicAddressesCompositionTest {
+
+    //------------------ Constructor Tests ------------------
+
+    @Test
+    public void constructor_nullMap_throwsAssertionError() {
+        // EP: null input
+        assertThrows(AssertionError.class, () ->
+            new PublicAddressesComposition(null));
+    }
+
+    @Test
+    public void constructor_mapWithEmptySet_throwsAssertionError() {
+        // EP: map with empty set
+        Map<Network, Set<PublicAddress>> addresses = new HashMap<>();
+        addresses.put(Network.BTC, new HashSet<>());
+
+        assertThrows(AssertionError.class, () ->
+            new PublicAddressesComposition(addresses));
+    }
```

Comment:
TZ & I think there shouldn't be tests for assertions since they can be disabled. Not 100% sure though

Reply:
i needed tests for assertions otherwise code coverage would be too low. Additionally, PublicAddressComposition should only be used by developers so assertions are mainly used rather than exceptions, but for redudency and more defensive coding, both assertions and exceptions shall can be used.

Reply:
Code coverage can be improved with assertCommandSuccess() function, which what other tests are using already